### PR TITLE
[log] optimize console log format

### PR DESCRIPTION
**Pull request by huof6829**
# Description
Please include a summary of the changes and the related issue. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes https://github.com/iotexproject/iotex-core/issues/3476

## Type of change
Please delete options that are not relevant.
- [] Bug fix (non-breaking change which fixes an issue)
- [] New feature (non-breaking change which adds functionality)
- [] Code refactor or improvement
- [] Breaking change (fix or feature that would cause a new or changed behavior of existing functionality)
- [] This change requires a documentation update

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration
- [] make test
- [] fullsync
- [] Other test (please specify)

**Test Configuration**:
- Firmware version:
- Hardware:
- Toolchain:
- SDK:

# Checklist:
- [] My code follows the style guidelines of this project
- [] I have performed a self-review of my code
- [] I have commented my code, particularly in hard-to-understand areas
- [] I have made corresponding changes to the documentation
- [] My changes generate no new warnings
- [] I have added tests that prove my fix is effective or that my feature works
- [] New and existing unit tests pass locally with my changes
- [] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by huof6829** [comment]
Just modify the starting log format for viewing. Leave the zap log because most codes use zap function. After zerolog merged, we can replace zap with zerolog.

output in cmdline:
stdout

/var/log/s.log is json formatter:
slog

**Comment by huof6829** [comment]
One note:    `"github.com/iotexproject/go-p2p"`  use `zap.Logger` now. If change to zerolog, this project should also change together.
https://github.com/iotexproject/iotex-core/blob/c75411c117fb90cf4f7fb01b755c8f9b245ec7ac/p2p/agent.go#L195

**Comment by dustinxie** [comment]
see comment in https://github.com/iotexproject/iotex-core/issues/3476, we need to achieve:
1. the output to log file is still same json format, for ELK log collection/analysis purpose
2. on top of that, if it's possible to make the output to commandline window better

replace with zerolog does not seem a good solution to achieve both

**Comment by millken** [comment]
zap `Encoder` supportted customize log format，you can implement an encoder to replace default [console encoder](https://github.com/uber-go/zap/blob/master/zapcore/console_encoder.go).

**Comment by huof6829** [comment]
> see comment in #3476, we need to achieve:
> 
> 1. the output to log file is still same json format, for ELK log collection/analysis purpose
> 2. on top of that, if it's possible to make the output to commandline window better
> 
> replace with zerolog does not seem a good solution to achieve both

the log file is json format. colored printing in cmmandline. see the picture above

**Comment by huof6829** [comment]
> see comment in #3476, we need to achieve:
> 
> 1. the output to log file is still same json format, for ELK log collection/analysis purpose
> 2. on top of that, if it's possible to make the output to commandline window better
> 
> replace with zerolog does not seem a good solution to achieve both

Because of ELK, the log file format can't change. I'll  try to find the humanly-printed setting by zap console encoder.

**Comment by raullenchai** [comment]
ping @dustinxie @CoderZhi @Liuhaai  - pls take another look

File: pkg/log/log.go [hunk 1]
```diff
@@ -85,19 +85,38 @@ func InitLoggers(globalCfg GlobalConfig, subCfgs map[string]GlobalConfig, opts .
...
+			zapcore.Lock(os.Stderr),
+			zap.InfoLevel))
+
+		coreOpt := zap.WrapCore(func(zapcore.Core) zapcore.Core {
+			return zapcore.NewTee(cores...)
+		})
+		var buildOpts []zap.Option
+		buildOpts = append(buildOpts, coreOpt)
+		buildOpts = append(buildOpts, opts...)
+		logger, err := cfg.Zap.Build(buildOpts...)
```
Comment (Liuhaai) [inline pkg/log/log.go:115]: `cfg.Zap.Build(coreOpt, opts)`?

**Review by Liuhaai — APPROVED** [review]

File: pkg/log/log.go (hunk 1 above)
Comment (huof6829) [reply pkg/log/log.go:115]: no，type of `opt` is []zap.Option

File: pkg/log/log.go [hunk 2]
```diff
@@ -85,19 +85,38 @@ func InitLoggers(globalCfg GlobalConfig, subCfgs map[string]GlobalConfig, opts .
...
+				cfg.Zap.Level))
 		}
+
+		consoleCfg := zap.NewDevelopmentConfig()
+		consoleCfg.EncoderConfig.EncodeLevel = zapcore.CapitalColorLevelEncoder
+		cores = append(cores, zapcore.NewCore(
+			zapcore.NewConsoleEncoder(consoleCfg.EncoderConfig),
+			zapcore.Lock(os.Stderr),
+			zap.InfoLevel))
+
```
Comment (dustinxie) [inline pkg/log/log.go:108]: so looks like add a new console encoder to process the output in human-friendly format?

i assume you already run locally to verify the log format in the file has remained the same?

Comment (huof6829) [reply pkg/log/log.go]: yes,  add console encoder.  I run in test-net, log format is the same with older

**Review by dustinxie — APPROVED** [review]

**Comment by raullenchai** [comment]
Thanks for making this one @huof6829 @dustinxie

**Pull request merged**